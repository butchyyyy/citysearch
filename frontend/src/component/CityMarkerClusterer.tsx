import React from "react"
import MarkerClusterer from "react-google-maps/lib/components/addons/MarkerClusterer"

interface Props {
  markerClusterer: boolean
  children?: any
}

const CityMarkerClusterer = (props: Props) => {
  if (props.markerClusterer) {
    return (
        <MarkerClusterer
            averageCenter
            enableRetinaIcons
            gridSize={60}
        >
          {props.children}
        </MarkerClusterer>
    )
  } else {
    return props.children
  }
}

export default CityMarkerClusterer
