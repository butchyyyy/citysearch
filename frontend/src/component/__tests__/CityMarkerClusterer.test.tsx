import CityMarkerClusterer from "component/CityMarkerClusterer"
import { shallow } from "enzyme"
import React from "react"

describe("Test CityMarkerClusterer Component", () => {
  const wrapper = shallow(<CityMarkerClusterer markerClusterer={false}>TestChildren</CityMarkerClusterer>)

  it("Test CityMarkerClusterer Snapshot", () => {
    expect(wrapper).toMatchSnapshot()
    // Enable marker clusters
    wrapper.setProps({ markerClusterer: true })
    expect(wrapper).toMatchSnapshot()
  })
})
