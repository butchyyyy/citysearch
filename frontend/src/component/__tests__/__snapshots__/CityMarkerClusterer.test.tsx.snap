// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test CityMarkerClusterer Component Test CityMarkerClusterer Snapshot 1`] = `"TestChildren"`;

exports[`Test CityMarkerClusterer Component Test CityMarkerClusterer Snapshot 2`] = `
<MarkerClusterer
  averageCenter={true}
  enableRetinaIcons={true}
  gridSize={60}
>
  TestChildren
</MarkerClusterer>
`;
